### PR TITLE
APIGOV-18557 - hide unnecessary config flags from help

### PR DESCRIPTION
**Pull request by dfeldick**

### CODE REVIEW

File: pkg/cmd/root.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func NewCmd(rootCmd *cobra.Command, exeName, desc string, initConfigHandler Init
 
 // Add the command line properties for the logger and path config
 func (c *agentRootCommand) addBaseProps() {
-	c.props.AddStringPersistentFlag(PathConfigFlag, ".", "Configuration file path for the agent")
+	c.props.AddStringPersistentFlag(PathConfigFlag, ".", "YAML configuration file path for the agent")
```
Comment (jcollins-axway) [inline pkg/cmd/root.go:138]: Is it confusing to say "YAML configuration.." When this is just the path to where the YAML file is, not the path including the YAML filename?

File: pkg/cmd/properties/properties.go [hunk 2]
```diff
@@ -120,6 +121,7 @@ func (p *properties) AddStringSliceProperty(name string, defaultVal []string, de
 		flagName := p.nameToFlagName(name)
 		p.rootCmd.Flags().StringSlice(flagName, defaultVal, description)
 		p.bindOrPanic(name, p.rootCmd.Flags().Lookup(flagName))
+		p.rootCmd.Flags().MarkHidden(flagName)
```
Comment (jcollins-axway) [inline pkg/cmd/properties/properties.go:124]: All properties are hidden?

Comment (dfeldick) [reply pkg/cmd/properties/properties.go:124]: Almost. The help looks like this now:
Flags:
      --envFile string      Path of the file with environment variables to override configuration
  -h, --help                help for discovery_agent
      --pathConfig string   YAML configuration file path for the agent
      --status              Get the status of all the Health Checks
      --synchronize         Run the sync process for the discovery agent
  -v, --version             version for discovery_agent

envFile, pathConfig, status, and synchronize were either creating flags (not props) or were creating persistent properties (don't know what that diff is). So, the methods I changed are not affecting these.

Comment (dfeldick) [reply pkg/cmd/properties/properties.go:124]: I toyed with passing a bool to hide/not hide to each of the methods, but that would bubble all over. Or, to copy/paste and have different methods completely to hide/not hide. I decided that for now my changes work.

File: pkg/cmd/root.go (hunk 1 above)
Comment (dfeldick) [reply pkg/cmd/root.go:138]: ah. Maybe "Path to the YAML configuration file for the agent"?

Comment (jcollins-axway) [reply pkg/cmd/root.go]: Path to directory containing YAML...?

Comment (dfeldick) [reply pkg/cmd/root.go]: ok

**Pull request merged**